src/transformers/generation/utils.py

```diff
@@ -2069,30 +2069,19 @@ def contrastive_search(
                     )
                     for key in all_outputs:
                         all_outputs[key].append(outputs[key])
+
+                    if self.config.is_encoder_decoder:
+                        next_hidden = outputs.decoder_hidden_states[-1]
+                        full_hidden_states = outputs.decoder_hidden_states
+
+                    else:
+                        next_hidden = outputs.hidden_states[-1]
+                        full_hidden_states = outputs.hidden_states
```

Comment:
These lines exit in both the if (low memory) and else (not low memory) code blocks, but note that they are not easily refactored because `next_hidden` and `full_hidden_states` must be returned iteratively for each `top_k` token when low memory is activated, but otherwise they are only returned once batch-wise.